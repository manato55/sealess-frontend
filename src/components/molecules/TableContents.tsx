import React from 'react'
import Tr from '../atoms/Tr'
import {Task} from '../../hooks/useReturnedTask';
import Table from '../atoms/Table'
import styled from 'styled-components'


interface Props {
    tasks: Task[];
    th: string[];
    pathName: string;
}

export const TableContents = (props: Props): React.ReactElement => {
    return (
        <>
            <Container>
                <Table>
                    <thead>
                        <tr>
                            {props.th.map((thName, index) =>
                                <th key={index}>{thName}</th>
                            )}
                        </tr>
                    </thead>
                    <tbody>
                        {props.tasks?.map((task,index) => 
                            <Tr
                                key={index}
                                task={task}
                                pathName={props.pathName}
                            />
                        )}
                    </tbody>
                </Table>
            </Container>
        </>
    )
}


const Container = styled.div`
    text-align: center;

    table {
        display: block;
        overflow-x: scroll;
        -webkit-overflow-scrolling: touch;
        text-align: center;
        table-layout: fixed;
        width: 100%;
        margin-top: 40px;
    }

    table tr, table td {
        white-space: nowrap;
        width: 10%;
    }
`;



export default TableContents