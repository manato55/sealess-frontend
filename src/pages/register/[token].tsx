import React, { useEffect, useState, useCallback } from 'react';
import { useRouter } from 'next/router';
import { useAuthenticate } from '../../hooks/useAuthenticate';
import Password from '../../components/molecules/Password';
import Button from '../../components/atoms/Button';
import AuthWrapper from '../../components/atoms/AuthWrapper';

export const Token = (): React.ReactElement => {
  const router = useRouter();
  const { tokenCheck } = useAuthenticate();
  const { officialRegistryForOrdinaryUser } = useAuthenticate();
  const [paramsToken, setParamsToken] = useState<any>(router.query.token);
  const [tokenChecker, setTokenChecker] = useState<boolean>(false);
  const [password, setPassword] = useState<string>('');

  useEffect(() => {
    const initialAction = async () => {
      const res = await tokenCheck(paramsToken);
      if (res) {
        setTokenChecker(true);
      }
    };
    initialAction();
  }, [paramsToken]);

  async function submitRegister(): Promise<void> {
    const data: {
      token: string;
      password: string;
    } = {
      token: paramsToken,
      password: password,
    };
    await officialRegistryForOrdinaryUser(data);
  }

  return (
    <>
      {tokenChecker ? (
        <AuthWrapper>
          <Password password={password} setPassword={setPassword} />
          <Button onClick={() => submitRegister()}>登録</Button>
        </AuthWrapper>
      ) : (
        'Loading...'
      )}
    </>
  );
};

export default Token;
