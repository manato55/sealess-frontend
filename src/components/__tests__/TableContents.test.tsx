import React from 'react';
import renderer from 'react-test-renderer';
import TableContents from '../molecules/TableContents';
import { RecoilRoot } from 'recoil';
import { mount, shallow } from 'enzyme';
import { render, fireEvent } from '@testing-library/react';
import 'jsdom-global/register';
import { renderHook, act } from '@testing-library/react-hooks';

describe('TableContents test', () => {
  test('snapshot', () => {
    const tree = renderer.create(<TableContents tasks={[]} th={[]} pathName={'test'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
