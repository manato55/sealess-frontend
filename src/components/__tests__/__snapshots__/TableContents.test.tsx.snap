// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableContents test snapshot 1`] = `
<div
  className="sc-gtsrHT kJNlq"
>
  <table
    className="sc-bdnxRM cZuGBU"
  >
    <thead>
      <tr />
    </thead>
    <tbody />
  </table>
</div>
`;
